### ISSUE TITLE
Failure to decode VARBINARY on Row.get(column)

### ISSUE THREAD

**Issue by mp911de** [mirromutth/r2dbc-mysql#62]
`Row.get(…)` fails when trying to decode a binary value.

**Stack trace**
```
 java.lang.IllegalArgumentException: Cannot decode value of type class java.lang.String for type 253
		at io.github.mirromutth.r2dbc.mysql.codec.DefaultCodecs.decode(DefaultCodecs.java:131)
		at io.github.mirromutth.r2dbc.mysql.MySqlRow.get0(MySqlRow.java:95)
		at io.github.mirromutth.r2dbc.mysql.MySqlRow.get(MySqlRow.java:63)
		at io.r2dbc.spi.Row.get(Row.java:127)
```

**Steps to reproduce**

```sql
CREATE TABLE certificates (cert VARBINARY(255));
INSERT INTO certificates VALUES(?);
SELECT cert FROM certificates;
```

```java
ConnectionFactory connectionFactory = …;

Mono.from(connectionFactory.create())
		.flatMapMany(it -> it.createStatement("CREATE TABLE certificates (cert VARBINARY(255))")
				.execute())
		.flatMap(Result::getRowsUpdated)
		.as(StepVerifier::create)
		.expectNextCount(1)
		.verifyComplete();

Mono.from(connectionFactory.create())
		.flatMapMany(it -> it.createStatement("INSERT INTO certificates VALUES(?)")
				.bind(0, new byte[]{1,2,3,4,5})
				.execute())
		.flatMap(Result::getRowsUpdated)
		.as(StepVerifier::create)
		.expectNext(1)
		.verifyComplete();

Mono.from(connectionFactory.create())
		.flatMapMany(it -> it.createStatement("SELECT cert FROM certificates")
				.execute())
		.flatMap(it -> it.map((row, rowMetadata) -> row.get("cert")))
		.as(StepVerifier::create)
		.expectNextCount(1)
		.verifyComplete();
```

**Comment by mirromutth** [comment]
This problem is because I mistakenly think of `VARSTRING`(id 253) as a string type, but it is equivalent to `VARBINARY`, a binary type.

And, the `Codecs` should check if the column metadata contains a valid charset. If it is invalid (i.e. binary), it is considered a binary type.

**Comment by mp911de** [comment]
While at it, please make sure that `BINARY` columns decode by default to `ByteBuffer` (see https://github.com/r2dbc/r2dbc-spi/blob/master/r2dbc-spec/src/main/asciidoc/data-types.adoc#datatypes.mapping.binary). I fixed that for Postgres. It also makes sense to allow retrieving and writing binary data as `byte[]`.

**Comment by mirromutth** [comment]
I have create a new issue for `ByteBuffer` decoding. I think they are different problems.

**Comment by mirromutth** [comment]
It solved by PR #64 , target version is 0.3.0.M3.
